src/app/modules/payments/__tests__/payments.service.spec.ts

```diff
@@ -155,7 +155,147 @@ describe('payments.service', () => {
+            description: 'some other description',
+            _id: new ObjectId(),
+          } as unknown as Product,
+          quantity: 1,
+          selected: true,
+        },
+        {
+          data: mockValidProduct,
+          quantity: 1,
+          selected: true,
+        },
+      ]
+
+      // Act
+      const result = PaymentsService.validatePaymentProducts(
+        mockValidProductsDefinition,
+        mockInvalidProductSubmission,
+      )
+
+      // Assert
+      expect(result.isErr()).toBeTrue()
+      expect(result._unsafeUnwrapErr()).toBeInstanceOf(
+        InvalidPaymentProductsError,
+      )
+      expect(result._unsafeUnwrapErr().message).toContain(
+        'There has been a change in the products available.',
+      )
+    })
+
+    it('should return with error if the name has changed', () => {
+      // Arrange
+      const mockInvalidProductSubmission: ProductItem[] = [
+        {
+          data: {
+            ...mockValidProduct,
+            name: 'some other name',
+            _id: new ObjectId(),
+          } as unknown as Product,
+          quantity: 1,
+          selected: true,
+        },
+        {
+          data: mockValidProduct,
+          quantity: 1,
+          selected: true,
+        },
+      ]
+
+      // Act
+      const result = PaymentsService.validatePaymentProducts(
+        mockValidProductsDefinition,
+        mockInvalidProductSubmission,
+      )
+
+      // Assert
+      expect(result.isErr()).toBeTrue()
+      expect(result._unsafeUnwrapErr()).toBeInstanceOf(
+        InvalidPaymentProductsError,
+      )
+      expect(result._unsafeUnwrapErr().message).toContain(
+        'There has been a change in the products available.',
+      )
+    })
+
+    it('should return with error if multi_qty has changed', () => {
+      // Arrange
+      const mockInvalidProductSubmission: ProductItem[] = [
+        {
+          data: {
+            ...mockValidProduct,
+            multi_qty: true,
+            _id: new ObjectId(),
+          } as unknown as Product,
+          quantity: 1,
+          selected: true,
+        },
+        {
+          data: mockValidProduct,
+          quantity: 1,
+          selected: true,
+        },
+      ]
+
+      // Act
+      const result = PaymentsService.validatePaymentProducts(
+        mockValidProductsDefinition,
+        mockInvalidProductSubmission,
+      )
+
+      // Assert
+      expect(result.isErr()).toBeTrue()
+      expect(result._unsafeUnwrapErr()).toBeInstanceOf(
+        InvalidPaymentProductsError,
+      )
+      expect(result._unsafeUnwrapErr().message).toContain(
+        'There has been a change in the products available.',
+      )
+    })
+
+    it('should return with error if the max_qty has changed', () => {
+      // Arrange
+      const mockInvalidProductSubmission: ProductItem[] = [
+        {
+          data: {
+            ...mockValidProduct,
+            max_qty: 5,
+            _id: new ObjectId(),
```

Comment:
Need to remove `_id` since we only want to test rejection from `max_qty` changed instead of `_id`

Reply:
👍  thanks for catching